diff -up tdesktop-3.3.0-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp.3~ tdesktop-3.3.0-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp
--- tdesktop-3.3.0-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp.3~	2021-10-31 09:27:25.000000000 +0100
+++ tdesktop-3.3.0-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp	2021-12-12 04:00:51.158515762 +0100
@@ -502,7 +502,7 @@ void HunspellService::readFile() {
 	auto &&zip = ranges::views::zip(
 		scripts, groupedWords
 	);
-	_addedWords = zip | ranges::to<WordsMap>();
+	_addedWords = zip | ranges::_to_::to<WordsMap>();
 
 }
 
diff -up tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_authorizations.cpp.3~ tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_authorizations.cpp
--- tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_authorizations.cpp.3~	2021-12-12 04:00:51.158515762 +0100
+++ tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_authorizations.cpp	2021-12-12 04:02:02.019851055 +0100
@@ -129,7 +129,7 @@ void Authorizations::reload() {
 				auths.vauthorizations().v
 			) | ranges::views::transform([](const MTPAuthorization &d) {
 				return ParseEntry(d.c_authorization());
-			}) | ranges::to<List>;
+			}) | ranges::_to_::to<List>;
 			_listChanges.fire({});
 		});
 	}).fail([=] {
diff -up tdesktop-3.3.0-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp.3~ tdesktop-3.3.0-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp
--- tdesktop-3.3.0-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp.3~	2021-12-08 05:35:03.000000000 +0100
+++ tdesktop-3.3.0-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp	2021-12-12 04:00:51.158515762 +0100
@@ -697,7 +697,7 @@ void Poll::updateAnswerVotes() {
 		ranges::views::all(
 			_poll->answers
 		) | ranges::views::transform(&PollAnswer::votes),
-		ranges::begin(VotesStorage));
+		ranges::_::begin(VotesStorage));
 
 	CountNicePercent(
 		gsl::make_span(VotesStorage).subspan(0, count),
diff -up tdesktop-3.3.0-full/Telegram/SourceFiles/ui/grouped_layout.cpp.3~ tdesktop-3.3.0-full/Telegram/SourceFiles/ui/grouped_layout.cpp
--- tdesktop-3.3.0-full/Telegram/SourceFiles/ui/grouped_layout.cpp.3~	2021-12-08 05:35:03.000000000 +0100
+++ tdesktop-3.3.0-full/Telegram/SourceFiles/ui/grouped_layout.cpp	2021-12-12 04:00:51.158515762 +0100
@@ -117,7 +117,7 @@ std::string Layouter::CountProportions(c
 		ratios
 	) | ranges::views::transform([](float64 ratio) {
 		return (ratio > 1.2) ? 'w' : (ratio < 0.8) ? 'n' : 'q';
-	}) | ranges::to<std::string>();
+	}) | ranges::_to_::to<std::string>();
 }
 
 std::vector<GroupMediaLayout> Layouter::layout() const {
