diff --git a/Telegram/CMakeLists.txt b/Telegram/CMakeLists.txt
index 911326b..87cf107 100644
--- a/Telegram/CMakeLists.txt
+++ b/Telegram/CMakeLists.txt
@@ -78,7 +78,7 @@ PRIVATE
     desktop-app::lib_stripe
     desktop-app::external_rlottie
     desktop-app::external_zlib
-    desktop-app::external_minizip
+    minizip
     desktop-app::external_kcoreaddons
     desktop-app::external_qt_static_plugins
     desktop-app::external_qt
diff --git a/cmake/external/CMakeLists.txt b/cmake/external/CMakeLists.txt
index ce7e67d..0e225c5 100644
--- a/cmake/external/CMakeLists.txt
+++ b/cmake/external/CMakeLists.txt
@@ -38,7 +38,6 @@ add_checked_subdirectory(jpeg)
 add_checked_subdirectory(kcoreaddons)
 add_checked_subdirectory(lz4)
 add_checked_subdirectory(microtex)
-add_checked_subdirectory(minizip)
 add_checked_subdirectory(openal)
 add_checked_subdirectory(openh264)
 add_checked_subdirectory(openssl)
