diff --git a/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp b/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
index 44f5acd..53de16c 100644
--- a/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
+++ b/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
@@ -268,7 +268,7 @@ void Create(Window::Notifications::System *system) {
 		Gio::DBusProxyFlags::NONE_,
 		kService,
 		kObjectPath,
-		[=](GObject::Object, Gio::AsyncResult res) {
+		[=](gi::repository::GObject::Object, Gio::AsyncResult res) {
 			auto result =
 				XdgNotifications::NotificationsProxy::new_for_bus_finish(res);
 
@@ -292,7 +292,7 @@ void Create(Window::Notifications::System *system) {
 			auto interface = XdgNotifications::Notifications(proxy);
 
 			interface.call_get_server_information([=](
-					GObject::Object,
+					gi::repository::GObject::Object,
 					Gio::AsyncResult res) mutable {
 				const auto result =
 					interface.call_get_server_information_finish(res);
@@ -317,7 +317,7 @@ void Create(Window::Notifications::System *system) {
 			});
 
 			interface.call_get_capabilities([=](
-					GObject::Object,
+					gi::repository::GObject::Object,
 					Gio::AsyncResult res) mutable {
 				const auto result = interface.call_get_capabilities_finish(
 					res);
@@ -741,7 +741,7 @@ void Manager::Private::showNotification(
 				const auto callbackWrap = gi::unwrap(
 					Gio::AsyncReadyCallback(
 						crl::guard(this, [=](
-								GObject::Object,
+								gi::repository::GObject::Object,
 								Gio::AsyncResult res) {
 							auto &sandbox = Core::Sandbox::Instance();
 							sandbox.customEnterFromEventLoop([&] {
diff --git a/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp b/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp
index 6b781fb..70a9a4c 100644
--- a/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp
+++ b/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp
@@ -99,7 +99,7 @@ SettingWatcher::SettingWatcher(
 		Gio::DBusProxyFlags::NONE_,
 		kService,
 		kObjectPath,
-		crl::guard(this, [=](GObject::Object, Gio::AsyncResult res) {
+		crl::guard(this, [=](gi::repository::GObject::Object, Gio::AsyncResult res) {
 			_private->interface = XdpSettings::Settings(
 				XdpSettings::SettingsProxy::new_for_bus_finish(res, nullptr));
 
