diff -up tdesktop-4.12.2-full/Telegram/SourceFiles/history/view/media/history_view_web_page.cpp.omv~ tdesktop-4.12.2-full/Telegram/SourceFiles/history/view/media/history_view_web_page.cpp
diff -up tdesktop-4.12.2-full/Telegram/SourceFiles/history/view/media/history_view_web_page.h.omv~ tdesktop-4.12.2-full/Telegram/SourceFiles/history/view/media/history_view_web_page.h
--- tdesktop-4.12.2-full/Telegram/SourceFiles/history/view/media/history_view_web_page.h.omv~	2023-12-01 18:23:14.991897011 +0100
+++ tdesktop-4.12.2-full/Telegram/SourceFiles/history/view/media/history_view_web_page.h	2023-12-01 18:28:55.001551006 +0100
@@ -131,6 +131,7 @@ private:
 	mutable std::unique_ptr<Ui::RippleAnimation> _ripple;
 
 	struct SponsoredData final {
+		SponsoredData() {}
 		PeerData *peer = nullptr;
 		Ui::PeerUserpicView userpicView;
 		QString buttonText;
