diff -up tdesktop-3.1.0-full/Telegram/CMakeLists.txt.5~ tdesktop-3.1.0-full/Telegram/CMakeLists.txt
--- tdesktop-3.1.0-full/Telegram/CMakeLists.txt.5~	2021-09-19 23:39:48.136038254 +0200
+++ tdesktop-3.1.0-full/Telegram/CMakeLists.txt	2021-09-19 23:40:28.722017216 +0200
@@ -1263,6 +1263,7 @@ else()
             desktop-app::external_dbusmenu_qt
             desktop-app::external_glibmm
             desktop-app::external_glib
+	    -ldl -ljpeg -lvpx -lopenh264
         )
     endif()
 
diff -up tdesktop-3.1.0-full/Telegram/lib_lottie/lottie/lottie_animation.cpp.5~ tdesktop-3.1.0-full/Telegram/lib_lottie/lottie/lottie_animation.cpp
diff -up tdesktop-3.1.0-full/Telegram/lib_lottie/lottie/lottie_icon.cpp.5~ tdesktop-3.1.0-full/Telegram/lib_lottie/lottie/lottie_icon.cpp
--- tdesktop-3.1.0-full/Telegram/lib_lottie/lottie/lottie_icon.cpp.5~	2021-09-18 22:36:35.000000000 +0200
+++ tdesktop-3.1.0-full/Telegram/lib_lottie/lottie/lottie_icon.cpp	2021-09-19 23:39:48.136038254 +0200
@@ -22,7 +22,7 @@ namespace {
 		const QByteArray &content,
 		QColor replacement) {
 	auto string = ReadUtf8(content);
-#ifndef DESKTOP_APP_USE_PACKAGED_RLOTTIE
+#if 0 //ndef DESKTOP_APP_USE_PACKAGED_RLOTTIE
 	auto list = std::vector<std::pair<std::uint32_t, std::uint32_t>>();
 	if (replacement != Qt::white) {
 		const auto value = (uint32_t(replacement.red()) << 16)
