diff -up tdesktop-6.5.0-full/Telegram/SourceFiles/data/data_media_types.cpp.3~ tdesktop-6.5.0-full/Telegram/SourceFiles/data/data_media_types.cpp
--- tdesktop-6.5.0-full/Telegram/SourceFiles/data/data_media_types.cpp.3~	2026-02-08 06:04:18.000000000 +0100
+++ tdesktop-6.5.0-full/Telegram/SourceFiles/data/data_media_types.cpp	2026-02-08 06:07:05.145610359 +0100
@@ -42,6 +42,7 @@ https://github.com/telegramdesktop/tdesk
 #include "dialogs/ui/dialogs_message_view.h"
 #include "ui/boxes/emoji_stake_box.h"
 #include "ui/image/image.h"
+#include "ui/effects/ripple_animation.h"
 #include "ui/effects/spoiler_mess.h"
 #include "ui/text/format_song_document_name.h"
 #include "ui/text/format_values.h"
diff -up tdesktop-6.5.0-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp.3~ tdesktop-6.5.0-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
--- tdesktop-6.5.0-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp.3~	2026-02-08 06:04:18.000000000 +0100
+++ tdesktop-6.5.0-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp	2026-02-08 06:05:39.040700436 +0100
@@ -31,6 +31,7 @@ https://github.com/telegramdesktop/tdesk
 #include "history/view/reactions/history_view_reactions_list.h"
 #include "info/info_memento.h"
 #include "info/profile/info_profile_widget.h"
+#include "ui/effects/ripple_animation.h"
 #include "ui/widgets/popup_menu.h"
 #include "ui/widgets/menu/menu_action.h"
 #include "ui/widgets/menu/menu_common.h"
