diff -up tdesktop-4.0.2-full/cmake/external/CMakeLists.txt.5~ tdesktop-4.0.2-full/cmake/external/CMakeLists.txt
--- tdesktop-4.0.2-full/cmake/external/CMakeLists.txt.5~	2022-07-06 00:34:18.376550280 +0200
+++ tdesktop-4.0.2-full/cmake/external/CMakeLists.txt	2022-07-06 00:34:59.853449373 +0200
@@ -42,7 +42,7 @@ if (LINUX AND NOT DESKTOP_APP_DISABLE_JE
 endif()
 add_checked_subdirectory(jpeg)
 add_checked_subdirectory(lz4)
-add_checked_subdirectory(minizip)
+#add_checked_subdirectory(minizip)
 if (LINUX)
     add_checked_subdirectory(nimf_qt5)
 endif()
diff -up tdesktop-4.0.2-full/Telegram/CMakeLists.txt.5~ tdesktop-4.0.2-full/Telegram/CMakeLists.txt
--- tdesktop-4.0.2-full/Telegram/CMakeLists.txt.5~	2022-07-06 00:34:18.356550169 +0200
+++ tdesktop-4.0.2-full/Telegram/CMakeLists.txt	2022-07-06 00:34:18.376550280 +0200
@@ -64,7 +64,7 @@ PRIVATE
     # include paths (usually ${PREFIX}/include/minizip) before any depend that
     # would add ${PREFIX}/include. This path may have a different <zip.h>,
     # for example installed by libzip (https://libzip.org).
-    desktop-app::external_minizip
+    minizip
 
     tdesktop::td_export
     tdesktop::td_mtproto
