diff -up tdesktop-6.3.4-full/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp.7~ tdesktop-6.3.4-full/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp
--- tdesktop-6.3.4-full/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp.7~	2025-11-26 13:44:09.000000000 +0100
+++ tdesktop-6.3.4-full/Telegram/SourceFiles/platform/linux/notifications_manager_linux.cpp	2025-11-26 17:25:41.188184578 +0100
@@ -262,7 +262,7 @@ void Create(Window::Notifications::Syste
 		Gio::DBusProxyFlags::NONE_,
 		kService,
 		kObjectPath,
-		[=](GObject::Object, Gio::AsyncResult res) {
+		[=](gi::repository::GObject::Object, Gio::AsyncResult res) {
 			auto result =
 				XdgNotifications::NotificationsProxy::new_for_bus_finish(res);
 
@@ -286,7 +286,7 @@ void Create(Window::Notifications::Syste
 			auto interface = XdgNotifications::Notifications(proxy);
 
 			interface.call_get_server_information([=](
-					GObject::Object,
+					gi::repository::GObject::Object,
 					Gio::AsyncResult res) mutable {
 				const auto result =
 					interface.call_get_server_information_finish(res);
@@ -311,7 +311,7 @@ void Create(Window::Notifications::Syste
 			});
 
 			interface.call_get_capabilities([=](
-					GObject::Object,
+					gi::repository::GObject::Object,
 					Gio::AsyncResult res) mutable {
 				const auto result = interface.call_get_capabilities_finish(
 					res);
@@ -750,7 +750,7 @@ void Manager::Private::showNotification(
 				const auto callbackWrap = gi::unwrap(
 					Gio::AsyncReadyCallback(
 						crl::guard(this, [=](
-								GObject::Object,
+								gi::repository::GObject::Object,
 								Gio::AsyncResult res) {
 							auto &sandbox = Core::Sandbox::Instance();
 							sandbox.customEnterFromEventLoop([&] {
diff -up tdesktop-6.3.4-full/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp.7~ tdesktop-6.3.4-full/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp
--- tdesktop-6.3.4-full/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp.7~	2025-11-09 05:14:42.000000000 +0100
+++ tdesktop-6.3.4-full/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp	2025-11-26 17:24:56.989835343 +0100
@@ -99,7 +99,7 @@ SettingWatcher::SettingWatcher(
 		Gio::DBusProxyFlags::NONE_,
 		kService,
 		kObjectPath,
-		crl::guard(this, [=](GObject::Object, Gio::AsyncResult res) {
+		crl::guard(this, [=](gi::repository::GObject::Object, Gio::AsyncResult res) {
 			_private->interface = XdpSettings::Settings(
 				XdpSettings::SettingsProxy::new_for_bus_finish(res, nullptr));
 
