diff -up tdesktop-6.9.3-full/Telegram/CMakeLists.txt.5~ tdesktop-6.9.3-full/Telegram/CMakeLists.txt
--- tdesktop-6.9.3-full/Telegram/CMakeLists.txt.5~	2026-06-15 02:15:18.258969473 +0200
+++ tdesktop-6.9.3-full/Telegram/CMakeLists.txt	2026-06-15 02:16:35.329996793 +0200
@@ -78,7 +78,7 @@ PRIVATE
     desktop-app::lib_stripe
     desktop-app::external_rlottie
     desktop-app::external_zlib
-    desktop-app::external_minizip
+    minizip
     desktop-app::external_kcoreaddons
     desktop-app::external_qt_static_plugins
     desktop-app::external_qt
diff -up tdesktop-6.9.3-full/cmake/external/CMakeLists.txt.5~ tdesktop-6.9.3-full/cmake/external/CMakeLists.txt
--- tdesktop-6.9.3-full/cmake/external/CMakeLists.txt.5~	2026-06-11 11:30:12.000000000 +0200
+++ tdesktop-6.9.3-full/cmake/external/CMakeLists.txt	2026-06-15 02:16:23.488804841 +0200
@@ -38,7 +38,6 @@ add_checked_subdirectory(jpeg)
 add_checked_subdirectory(kcoreaddons)
 add_checked_subdirectory(lz4)
 add_checked_subdirectory(microtex)
-add_checked_subdirectory(minizip)
 add_checked_subdirectory(openal)
 add_checked_subdirectory(openh264)
 add_checked_subdirectory(openssl)
