diff -up tdesktop-4.8.4-full/Telegram/lib_base/base/algorithm.h.omv~ tdesktop-4.8.4-full/Telegram/lib_base/base/algorithm.h
--- tdesktop-4.8.4-full/Telegram/lib_base/base/algorithm.h.omv~	2023-07-13 23:32:07.626963330 +0200
+++ tdesktop-4.8.4-full/Telegram/lib_base/base/algorithm.h	2023-07-13 23:32:13.722022417 +0200
@@ -6,6 +6,7 @@
 //
 #pragma once
 
+#include <QtCore/QString>
 #include <QtCore/QLatin1String>
 #include <memory>
 
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/passcode_box.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/passcode_box.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/passcode_box.cpp.omv~	2023-07-14 00:22:01.793389059 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/passcode_box.cpp	2023-07-14 00:22:50.050893997 +0200
@@ -685,12 +685,12 @@ void PasscodeBox::save(bool force) {
 		}
 		if (!onlyCheck && !_recoverEmail->isHidden() && email.isEmpty() && !force) {
 			_skipEmailWarning = true;
-			_replacedBy = getDelegate()->show(Ui::MakeConfirmBox({
+			_replacedBy = static_cast<Ui::BoxContent*>(getDelegate()->show(Ui::MakeConfirmBox({
 				.text = { tr::lng_cloud_password_about_recover() },
 				.confirmed = crl::guard(this, [this] { save(true); }),
 				.confirmText = tr::lng_cloud_password_skip_email(),
 				.confirmStyle = &st::attentionBoxButton,
-			}));
+			})));
 		} else if (onlyCheck) {
 			submitOnlyCheckCloudPassword(old);
 		} else if (_oldPasscode->isHidden()) {
@@ -1118,7 +1118,7 @@ void PasscodeBox::recover() {
 		recoverExpired();
 	}, lifetime());
 
-	_replacedBy = box;
+	_replacedBy = static_cast<Ui::BoxContent*>(box);
 }
 
 void PasscodeBox::recoverStarted(const MTPauth_PasswordRecovery &result) {
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/peers/edit_participant_box.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/peers/edit_participant_box.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/peers/edit_participant_box.cpp.omv~	2023-07-13 23:57:20.386482811 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/peers/edit_participant_box.cpp	2023-07-14 00:09:32.316028710 +0200
@@ -391,14 +391,14 @@ void EditAdminBox::prepare() {
 				const auto phrase = peer()->isBroadcast()
 					? tr::lng_bot_sure_add_text_channel
 					: tr::lng_bot_sure_add_text_group;
-				_confirmBox = getDelegate()->show(Ui::MakeConfirmBox({
+				_confirmBox = static_cast<Ui::BoxContent*>(getDelegate()->show(Ui::MakeConfirmBox({
 					phrase(
 						tr::now,
 						lt_group,
 						Ui::Text::Bold(peer()->name()),
 						Ui::Text::WithEntities),
 					crl::guard(this, [=] { finishAddAdmin(); })
-				}));
+				})));
 			} else {
 				_finishSave();
 			}
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/peers/edit_participant_box.h.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/peers/edit_participant_box.h
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/phone_banned_box..c.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/phone_banned_box..c
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/phone_banned_box.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/phone_banned_box.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/phone_banned_box.cpp.omv~	2023-07-14 00:25:32.714021795 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/phone_banned_box.cpp	2023-07-14 00:25:47.274205226 +0200
@@ -55,7 +55,7 @@ void ShowPhoneBannedError(
 			(*box)->closeBox();
 		}
 	};
-	*box = controller->show(
+	*box = static_cast<Ui::BoxContent*>(controller->show(
 		Ui::MakeConfirmBox({
 			.text = tr::lng_signin_banned_text(),
 			.cancelled = [=](Fn<void()> &&close) {
@@ -66,7 +66,7 @@ void ShowPhoneBannedError(
 			.cancelText = tr::lng_signin_banned_help(),
 			.strictCancel = true,
 		}),
-		Ui::LayerOption::CloseOther);
+		Ui::LayerOption::CloseOther));
 }
 
 } // namespace Ui
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/sessions_box.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/sessions_box.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/sessions_box.cpp.omv~	2023-07-14 00:26:36.769821344 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/sessions_box.cpp	2023-07-14 00:26:45.905933874 +0200
@@ -845,7 +845,7 @@ void SessionsContent::terminate(Fn<void(
 		.confirmText = tr::lng_settings_reset_button(),
 		.confirmStyle = &st::attentionBoxButton,
 	});
-	_terminateBox = Ui::MakeWeak(box.data());
+	_terminateBox = static_cast<Ui::BoxContent*>(Ui::MakeWeak(box.data()));
 	_controller->show(std::move(box));
 }
 
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/url_auth_box.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/url_auth_box.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/url_auth_box.cpp.omv~	2023-07-14 00:28:11.317969446 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/boxes/url_auth_box.cpp	2023-07-14 00:30:19.927480175 +0200
@@ -171,9 +171,9 @@ void UrlAuthBox::Request(
 			}).send();
 		}
 	};
-	*box = Ui::show(
+	*box = static_cast<Ui::BoxContent*>(Ui::show(
 		Box<UrlAuthBox>(session, url, qs(request.vdomain()), bot, callback),
-		Ui::LayerOption::KeepOther);
+		Ui::LayerOption::KeepOther));
 }
 
 void UrlAuthBox::Request(
@@ -222,9 +222,9 @@ void UrlAuthBox::Request(
 			}).send();
 		}
 	};
-	*box = Ui::show(
+	*box = static_cast<Ui::BoxContent*>(Ui::show(
 		Box<UrlAuthBox>(session, url, qs(request.vdomain()), bot, callback),
-		Ui::LayerOption::KeepOther);
+		Ui::LayerOption::KeepOther));
 }
 
 UrlAuthBox::UrlAuthBox(
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/calls/group/calls_group_rtmp.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/calls/group/calls_group_rtmp.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/calls/group/calls_group_rtmp.cpp.omv~	2023-07-14 00:32:13.905779583 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/calls/group/calls_group_rtmp.cpp	2023-07-14 00:32:24.781901939 +0200
@@ -193,7 +193,7 @@ void StartRtmpProcess::createBox() {
 	) | rpl::start_with_next([=] {
 		_request = nullptr;
 	}, _request->lifetime);
-	_request->box = Ui::MakeWeak(object.data());
+	_request->box = static_cast<Ui::BoxContent*>(Ui::MakeWeak(object.data()));
 	_request->show->showBox(std::move(object));
 }
 
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp:7.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp:7
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp.omv~	2023-07-14 00:43:13.766877991 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp	2023-07-14 00:43:29.532042305 +0200
@@ -771,8 +771,8 @@ void Application::badMtprotoConfiguratio
 				settings().proxy().selected(),
 				MTP::ProxyData::Settings::System);
 		};
-		_badProxyDisableBox = Ui::show(
-			Ui::MakeInformBox(Lang::Hard::ProxyConfigError()));
+		_badProxyDisableBox = static_cast<Ui::BoxContent*>(Ui::show(
+			Ui::MakeInformBox(Lang::Hard::ProxyConfigError())));
 		_badProxyDisableBox->boxClosing(
 		) | rpl::start_with_next(
 			disableCallback,
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp:.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/core/application.cpp:
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/passport/passport_panel_controller.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/passport/passport_panel_controller.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/passport/passport_panel_controller.cpp.omv~	2023-07-14 00:47:03.818262643 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/passport/passport_panel_controller.cpp	2023-07-14 00:47:20.616435821 +0200
@@ -1431,11 +1431,11 @@ void PanelController::cancelEditScope()
 
 	if (_panelHasUnsavedChanges && _panelHasUnsavedChanges()) {
 		if (!_confirmForgetChangesBox) {
-			_confirmForgetChangesBox = show(Ui::MakeConfirmBox({
+			_confirmForgetChangesBox = static_cast<Ui::BoxContent*>(show(Ui::MakeConfirmBox({
 				.text = tr::lng_passport_sure_cancel(),
 				.confirmed = [=] { _panel->showForm(); },
 				.confirmText = tr::lng_continue(),
-			}));
+			})));
 			_editScopeBoxes.emplace_back(_confirmForgetChangesBox);
 		}
 	} else {
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/window/window_controller.cpp:27.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/window/window_controller.cpp:27
diff -up tdesktop-4.8.4-full/Telegram/SourceFiles/window/window_controller.cpp.omv~ tdesktop-4.8.4-full/Telegram/SourceFiles/window/window_controller.cpp
--- tdesktop-4.8.4-full/Telegram/SourceFiles/window/window_controller.cpp.omv~	2023-07-14 00:59:37.828718961 +0200
+++ tdesktop-4.8.4-full/Telegram/SourceFiles/window/window_controller.cpp	2023-07-14 00:59:52.001916911 +0200
@@ -272,7 +272,7 @@ void Controller::checkLockByTerms() {
 		crl::on_main(widget(), [=] { checkLockByTerms(); });
 	});
 
-	_termsBox = box;
+	_termsBox = static_cast<Ui::BoxContent*>(box);
 }
 
 void Controller::showTermsDecline() {
