diff --git a/Telegram/SourceFiles/core/utils.cpp b/Telegram/SourceFiles/core/utils.cpp
index 3a924059..53b21c63 100644
--- a/Telegram/SourceFiles/core/utils.cpp
+++ b/Telegram/SourceFiles/core/utils.cpp
@@ -175,7 +175,7 @@ namespace ThirdParty {
 
 	void finish() {
 		CRYPTO_cleanup_all_ex_data();
-#ifndef LIBRESSL_VERSION_NUMBER
+#if !defined(LIBRESSL_VERSION_NUMBER) && OPENSSL_VERSION_MAJOR < 3
 		FIPS_mode_set(0);
 #endif
 		ENGINE_cleanup();
