diff --git a/Telegram/lib_ui/ui/emoji_config.cpp b/Telegram/lib_ui/ui/emoji_config.cpp
index f92235f..3ec2c99 100644
--- a/Telegram/lib_ui/ui/emoji_config.cpp
+++ b/Telegram/lib_ui/ui/emoji_config.cpp
@@ -25,6 +25,8 @@
 
 #include <crl/crl_async.h>
 
+#include <iostream>
+
 namespace Ui {
 namespace Emoji {
 namespace {
@@ -445,7 +447,8 @@ void UniversalImages::draw(
 
 QImage UniversalImages::generate(int size, int index) const {
 	Expects(size > 0);
-	Expects(index < _sprites.size());
+	if(index >= _sprites.size())
+		return QImage();
 
 	const auto rows = RowsCount(index);
 	const auto large = kUniversalSize;
