diff -up tdesktop-4.16.4-full/Telegram/SourceFiles/data/data_media_types.cpp.omv~ tdesktop-4.16.4-full/Telegram/SourceFiles/data/data_media_types.cpp
--- tdesktop-4.16.4-full/Telegram/SourceFiles/data/data_media_types.cpp.omv~	2024-04-08 15:37:12.591794021 +0200
+++ tdesktop-4.16.4-full/Telegram/SourceFiles/data/data_media_types.cpp	2024-04-08 15:37:21.551865120 +0200
@@ -35,6 +35,7 @@ https://github.com/telegramdesktop/tdesk
 #include "history/view/media/history_view_userpic_suggestion.h"
 #include "dialogs/ui/dialogs_message_view.h"
 #include "ui/image/image.h"
+#include "ui/effects/ripple_animation.h"
 #include "ui/effects/spoiler_mess.h"
 #include "ui/text/format_song_document_name.h"
 #include "ui/text/format_values.h"
diff -up tdesktop-4.16.4-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp.omv~ tdesktop-4.16.4-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp
--- tdesktop-4.16.4-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp.omv~	2024-04-08 15:46:10.168249539 +0200
+++ tdesktop-4.16.4-full/Telegram/SourceFiles/history/view/history_view_context_menu.cpp	2024-04-08 15:46:20.824340615 +0200
@@ -28,6 +28,7 @@ https://github.com/telegramdesktop/tdesk
 #include "history/view/reactions/history_view_reactions_list.h"
 #include "info/info_memento.h"
 #include "info/profile/info_profile_widget.h"
+#include "ui/effects/ripple_animation.h"
 #include "ui/widgets/popup_menu.h"
 #include "ui/widgets/menu/menu_action.h"
 #include "ui/widgets/menu/menu_common.h"
diff -up tdesktop-4.16.4-full/Telegram/SourceFiles/stdafx.h.omv~ tdesktop-4.16.4-full/Telegram/SourceFiles/stdafx.h
