diff -up tdesktop-2.9.12-full/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp.7~ tdesktop-2.9.12-full/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp
--- tdesktop-2.9.12-full/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp.7~	2021-08-24 20:08:35.000000000 +0200
+++ tdesktop-2.9.12-full/Telegram/SourceFiles/media/streaming/media_streaming_video_track.cpp	2021-08-25 15:52:31.571794564 +0200
@@ -15,9 +15,11 @@ https://github.com/telegramdesktop/tdesk
 #include "zlib.h"
 #include <cfenv>
 
+#ifdef __MSVC__
 extern "C" {
 extern int __isa_available;
 }
+#endif
 
 #define TO_LOG(x) debugLog(QString x)
 
@@ -229,7 +231,10 @@ VideoTrackObject::VideoTrackObject(
 	Expects(_error != nullptr);
 
 	TO_LOG(("created,speed:%1,mode:%2,position:%3,sync:%4,"
-		"loop:%5,wait:%6,duration:%7,initialized:%8,isa:%9"
+		"loop:%5,wait:%6,duration:%7,initialized:%8"
+#ifdef __MSVC__
+		",isa:%9"
+#endif
 		).arg(options.speed
 		).arg(int(options.mode)
 		).arg(options.position
@@ -238,7 +243,10 @@ VideoTrackObject::VideoTrackObject(
 		).arg(options.waitForMarkAsShown ? "true" : "false"
 		).arg(_stream.duration
 		).arg(_shared->initialized() ? "true" : "false"
-		).arg(__isa_available));
+#ifdef __MSVC__
+		).arg(__isa_available
+#endif
+		));
 }
 
 rpl::producer<> VideoTrackObject::checkNextFrame() const {
