diff -up tdesktop-6.7.5-full/Telegram/CMakeLists.txt.4~ tdesktop-6.7.5-full/Telegram/CMakeLists.txt
--- tdesktop-6.7.5-full/Telegram/CMakeLists.txt.4~	2026-04-07 18:19:25.000000000 +0200
+++ tdesktop-6.7.5-full/Telegram/CMakeLists.txt	2026-04-08 13:04:36.170591780 +0200
@@ -95,6 +95,7 @@ PRIVATE
     desktop-app::external_openssl
     desktop-app::external_openal
     desktop-app::external_xxhash
+    -ljpeg -lvpx -lopenh264
 )
 
 if (NOT DESKTOP_APP_DISABLE_SWIFT6)
