diff --git a/Telegram/SourceFiles/core/crash_report_window.cpp b/Telegram/SourceFiles/core/crash_report_window.cpp
index 07782ef..137fd03 100644
--- a/Telegram/SourceFiles/core/crash_report_window.cpp
+++ b/Telegram/SourceFiles/core/crash_report_window.cpp
@@ -28,6 +28,8 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include <QtCore/QStandardPaths>
 #include <QtCore/QTimer>
 
+#define Z_DEFAULT_COMPRESSION (-1)
+
 namespace {
 
 constexpr auto kDefaultProxyPort = 80;
diff --git a/Telegram/SourceFiles/payments/ui/payments_panel.cpp b/Telegram/SourceFiles/payments/ui/payments_panel.cpp
index eac1cf9..c9cff01 100644
--- a/Telegram/SourceFiles/payments/ui/payments_panel.cpp
+++ b/Telegram/SourceFiles/payments/ui/payments_panel.cpp
@@ -26,6 +26,7 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 #include "webview/webview_interface.h"
 #include "styles/style_payments.h"
 #include "styles/style_layers.h"
+#include <QJsonDocument>
 
 namespace Payments::Ui {
 namespace {
diff --git a/Telegram/SourceFiles/window/themes/window_theme_editor_box.cpp b/Telegram/SourceFiles/window/themes/window_theme_editor_box.cpp
index b458a0d..7b97ed8 100644
--- a/Telegram/SourceFiles/window/themes/window_theme_editor_box.cpp
+++ b/Telegram/SourceFiles/window/themes/window_theme_editor_box.cpp
@@ -49,6 +49,10 @@ https://github.com/telegramdesktop/tdesktop/blob/master/LEGAL
 
 #include <QtCore/QBuffer>
 
+#ifndef Z_DEFAULT_COMPRESSION
+#define Z_DEFAULT_COMPRESSION (-1)
+#endif
+
 namespace Window {
 namespace Theme {
 namespace {
diff --git a/Telegram/lib_base/base/zlib_help.h b/Telegram/lib_base/base/zlib_help.h
index 59d70fd..f78e21f 100644
--- a/Telegram/lib_base/base/zlib_help.h
+++ b/Telegram/lib_base/base/zlib_help.h
@@ -15,6 +15,11 @@
 #undef small
 #endif // small
 
+typedef void* voidp;
+typedef void* voidpf;
+typedef unsigned int uInt;
+typedef unsigned long uLong;
+
 namespace zlib {
 namespace internal {
 
