diff -up tdesktop-6.1.3-full/Telegram/ThirdParty/dispatch/src/event/event_epoll.c.omv~ tdesktop-6.1.3-full/Telegram/ThirdParty/dispatch/src/event/event_epoll.c
--- tdesktop-6.1.3-full/Telegram/ThirdParty/dispatch/src/event/event_epoll.c.omv~	2025-09-23 21:50:33.732341536 +0200
+++ tdesktop-6.1.3-full/Telegram/ThirdParty/dispatch/src/event/event_epoll.c	2025-09-23 21:50:40.868745523 +0200
@@ -89,7 +89,7 @@ DISPATCH_ALWAYS_INLINE
 static inline uint32_t
 _dispatch_muxnote_armed_events(dispatch_muxnote_t dmn)
 {
-	return dmn->dmn_events & ~dmn->dmn_disarmed_events;
+	return dmn->dmn_events & (uint32_t)~dmn->dmn_disarmed_events;
 }
 
 DISPATCH_ALWAYS_INLINE
diff -up tdesktop-6.1.3-full/Telegram/lib_ui/CMakeLists.txt.omv~ tdesktop-6.1.3-full/Telegram/lib_ui/CMakeLists.txt
diff -up tdesktop-6.1.3-full/Telegram/lib_ui/ui/widgets/buttons.h.omv~ tdesktop-6.1.3-full/Telegram/lib_ui/ui/widgets/buttons.h
--- tdesktop-6.1.3-full/Telegram/lib_ui/ui/widgets/buttons.h.omv~	2025-09-23 23:06:48.632993713 +0200
+++ tdesktop-6.1.3-full/Telegram/lib_ui/ui/widgets/buttons.h	2025-09-23 23:06:55.042828630 +0200
@@ -276,7 +276,7 @@ public:
 		const Text::MarkedContext &context = {});
 	SettingsButton(
 		QWidget *parent,
-		nullptr_t,
+		std::nullptr_t,
 		const style::SettingsButton &st = st::defaultSettingsButton);
 	~SettingsButton();
 
diff -up tdesktop-6.1.3-full/Telegram/SourceFiles/platform/linux/integration_linux.cpp.omv~ tdesktop-6.1.3-full/Telegram/SourceFiles/platform/linux/integration_linux.cpp
--- tdesktop-6.1.3-full/Telegram/SourceFiles/platform/linux/integration_linux.cpp.omv~	2025-09-25 01:48:10.855627410 +0200
+++ tdesktop-6.1.3-full/Telegram/SourceFiles/platform/linux/integration_linux.cpp	2025-09-25 01:48:19.715897742 +0200
@@ -18,6 +18,7 @@ https://github.com/telegramdesktop/tdesk
 #include "base/random.h"
 
 #include <QtCore/QAbstractEventDispatcher>
+#include <QElapsedTimer>
 
 #include <gio/gio.hpp>
 #include <xdpinhibit/xdpinhibit.hpp>
