diff -up tdesktop-2.8.5-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp.omv~ tdesktop-2.8.5-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp
--- tdesktop-2.8.5-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp.omv~	2021-07-06 16:44:12.521634900 +0200
+++ tdesktop-2.8.5-full/Telegram/lib_spellcheck/spellcheck/third_party/hunspell_controller.cpp	2021-07-06 16:44:21.498125419 +0200
@@ -502,7 +502,7 @@ void HunspellService::readFile() {
 	auto &&zip = ranges::views::zip(
 		scripts, groupedWords
 	);
-	_addedWords = zip | ranges::to<WordsMap>();
+	_addedWords = zip | ranges::_to_::to<WordsMap>();
 
 }
 
diff -up tdesktop-2.8.5-full/Telegram/SourceFiles/api/api_authorizations.cpp.omv~ tdesktop-2.8.5-full/Telegram/SourceFiles/api/api_authorizations.cpp
--- tdesktop-2.8.5-full/Telegram/SourceFiles/api/api_authorizations.cpp.omv~	2021-07-06 18:01:02.725980254 +0200
+++ tdesktop-2.8.5-full/Telegram/SourceFiles/api/api_authorizations.cpp	2021-07-06 18:01:07.542732055 +0200
@@ -109,7 +109,7 @@ void Authorizations::reload() {
 				auths.vauthorizations().v
 			) | ranges::views::transform([](const MTPAuthorization &d) {
 				return ParseEntry(d.c_authorization());
-			}) | ranges::to<List>;
+			}) | ranges::_to_::to<List>;
 			_listChanges.fire({});
 		});
 	}).fail([=](const MTP::Error &error) {
diff -up tdesktop-2.8.5-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp.omv~ tdesktop-2.8.5-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp
--- tdesktop-2.8.5-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp.omv~	2021-07-06 21:52:51.871643891 +0200
+++ tdesktop-2.8.5-full/Telegram/SourceFiles/history/view/media/history_view_poll.cpp	2021-07-06 21:52:56.448380092 +0200
@@ -700,7 +700,7 @@ void Poll::updateAnswerVotes() {
 		ranges::views::all(
 			_poll->answers
 		) | ranges::views::transform(&PollAnswer::votes),
-		ranges::begin(VotesStorage));
+		ranges::_::begin(VotesStorage));
 
 	CountNicePercent(
 		gsl::make_span(VotesStorage).subspan(0, count),
diff -up tdesktop-2.8.5-full/Telegram/SourceFiles/ui/grouped_layout.cpp.omv~ tdesktop-2.8.5-full/Telegram/SourceFiles/ui/grouped_layout.cpp
--- tdesktop-2.8.5-full/Telegram/SourceFiles/ui/grouped_layout.cpp.omv~	2021-07-06 16:37:30.585891475 +0200
+++ tdesktop-2.8.5-full/Telegram/SourceFiles/ui/grouped_layout.cpp	2021-07-06 16:43:29.332867397 +0200
@@ -117,7 +117,7 @@ std::string Layouter::CountProportions(c
 		ratios
 	) | ranges::views::transform([](float64 ratio) {
 		return (ratio > 1.2) ? 'w' : (ratio < 0.8) ? 'n' : 'q';
-	}) | ranges::to<std::string>();
+	}) | ranges::_to_::to<std::string>();
 }
 
 std::vector<GroupMediaLayout> Layouter::layout() const {
