diff -up tdesktop-4.9.3-full/Telegram/SourceFiles/settings/settings_websites.cpp.omv~ tdesktop-4.9.3-full/Telegram/SourceFiles/settings/settings_websites.cpp
--- tdesktop-4.9.3-full/Telegram/SourceFiles/settings/settings_websites.cpp.omv~	2023-08-23 13:31:21.518939843 +0200
+++ tdesktop-4.9.3-full/Telegram/SourceFiles/settings/settings_websites.cpp	2023-08-23 13:31:47.755142907 +0200
@@ -530,7 +530,7 @@ void Content::terminate(
 			*block = box->addRow(object_ptr<Ui::Checkbox>(box, blockText));
 		}
 	});
-	_terminateBox = Ui::MakeWeak(box.data());
+	_terminateBox = Ui::MakeWeak(static_cast<Ui::BoxContent*>(box.data()));
 	_controller->show(std::move(box));
 }
 
