diff -up tdesktop-2.8.6-full/Telegram/SourceFiles/api/api_media.cpp.omv~ tdesktop-2.8.6-full/Telegram/SourceFiles/api/api_media.cpp
--- tdesktop-2.8.6-full/Telegram/SourceFiles/api/api_media.cpp.omv~	2021-07-08 21:37:06.088688751 +0200
+++ tdesktop-2.8.6-full/Telegram/SourceFiles/api/api_media.cpp	2021-07-08 21:37:18.492225518 +0200
@@ -82,7 +82,7 @@ MTPInputMedia PrepareUploadedPhoto(
 	return MTP_inputMediaUploadedPhoto(
 		MTP_flags(flags),
 		file,
-		MTP_vector<MTPInputDocument>(ranges::to<QVector>(attachedStickers)),
+		MTP_vector<MTPInputDocument>(ranges::_to_::to<QVector>(attachedStickers)),
 		MTP_int(0));
 }
 
@@ -107,7 +107,7 @@ MTPInputMedia PrepareUploadedDocument(
 		thumb.value_or(MTPInputFile()),
 		MTP_string(document->mimeString()),
 		ComposeSendingDocumentAttributes(document),
-		MTP_vector<MTPInputDocument>(ranges::to<QVector>(attachedStickers)),
+		MTP_vector<MTPInputDocument>(ranges::_to_::to<QVector>(attachedStickers)),
 		MTP_int(0));
 }
 
