diff -up tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_media.cpp.1~ tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_media.cpp
--- tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_media.cpp.1~	2021-12-12 03:59:29.589140809 +0100
+++ tdesktop-3.3.0-full/Telegram/SourceFiles/api/api_media.cpp	2021-12-12 04:00:30.742420743 +0100
@@ -83,7 +83,7 @@ MTPInputMedia PrepareUploadedPhoto(Remot
 		MTP_flags(flags),
 		info.file,
 		MTP_vector<MTPInputDocument>(
-			ranges::to<QVector>(info.attachedStickers)),
+			ranges::_to_::to<QVector>(info.attachedStickers)),
 		MTP_int(0));
 }
 
@@ -107,7 +107,7 @@ MTPInputMedia PrepareUploadedDocument(
 		MTP_string(document->mimeString()),
 		ComposeSendingDocumentAttributes(document),
 		MTP_vector<MTPInputDocument>(
-			ranges::to<QVector>(info.attachedStickers)),
+			ranges::_to_::to<QVector>(info.attachedStickers)),
 		MTP_int(0));
 }
 
