diff -up tdesktop-1.9.22-full/Telegram/lib_base/base/thread_safe_wrap.h.omv~ tdesktop-1.9.22-full/Telegram/lib_base/base/thread_safe_wrap.h
--- tdesktop-1.9.22-full/Telegram/lib_base/base/thread_safe_wrap.h.omv~	2020-03-31 02:20:10.672196685 +0200
+++ tdesktop-1.9.22-full/Telegram/lib_base/base/thread_safe_wrap.h	2020-03-31 02:23:13.367727858 +0200
@@ -7,6 +7,8 @@
 #pragma once
 
 #include <utility>
+#include <QMutex>
+#include <QMutexLocker>
 
 namespace base {
 
@@ -31,7 +33,7 @@ public:
 
 private:
 	T _value;
-	QMutex _mutex;
+	mutable QMutex _mutex;
 
 };
 
diff -up tdesktop-1.9.22-full/Telegram/SourceFiles/calls/calls_call.cpp.omv~ tdesktop-1.9.22-full/Telegram/SourceFiles/calls/calls_call.cpp
--- tdesktop-1.9.22-full/Telegram/SourceFiles/calls/calls_call.cpp.omv~	2020-03-31 02:19:45.738122726 +0200
+++ tdesktop-1.9.22-full/Telegram/SourceFiles/calls/calls_call.cpp	2020-03-31 02:19:52.045141471 +0200
@@ -583,7 +583,7 @@ void Call::createAndStartController(cons
 #else // Q_OS_WIN
 		const auto callLogUtf = QFile::encodeName(callLogNative);
 		config.logFilePath.resize(callLogUtf.size());
-		ranges::copy(callLogUtf, config.logFilePath.begin());
+		::ranges::copy(callLogUtf, config.logFilePath.begin());
 #endif // Q_OS_WIN
 		QFile(callLogPath).remove();
 		QDir().mkpath(callLogFolder);
diff -up tdesktop-1.9.22-full/Telegram/SourceFiles/settings/settings_calls.cpp.omv~ tdesktop-1.9.22-full/Telegram/SourceFiles/settings/settings_calls.cpp
--- tdesktop-1.9.22-full/Telegram/SourceFiles/settings/settings_calls.cpp.omv~	2020-03-31 02:24:18.196912538 +0200
+++ tdesktop-1.9.22-full/Telegram/SourceFiles/settings/settings_calls.cpp	2020-03-31 02:24:39.026971524 +0200
@@ -76,7 +76,7 @@ void Calls::setupContent(not_null<Window
 			return tr::lng_settings_call_device_default(tr::now);
 		}
 		const auto &list = VoIP::EnumerateAudioOutputs();
-		const auto i = ranges::find(
+		const auto i = ::ranges::find(
 			list,
 			Global::CallOutputDeviceID(),
 			getId);
@@ -90,7 +90,7 @@ void Calls::setupContent(not_null<Window
 			return tr::lng_settings_call_device_default(tr::now);
 		}
 		const auto &list = VoIP::EnumerateAudioInputs();
-		const auto i = ranges::find(
+		const auto i = ::ranges::find(
 			list,
 			Global::CallInputDeviceID(),
 			getId);
@@ -112,11 +112,11 @@ void Calls::setupContent(not_null<Window
 		st::settingsButton
 	)->addClickHandler([=] {
 		const auto &devices = VoIP::EnumerateAudioOutputs();
-		const auto options = ranges::view::concat(
-			ranges::view::single(tr::lng_settings_call_device_default(tr::now)),
-			devices | ranges::view::transform(getName)
-		) | ranges::to_vector;
-		const auto i = ranges::find(
+		const auto options = ::ranges::view::concat(
+			::ranges::view::single(tr::lng_settings_call_device_default(tr::now)),
+			devices | ::ranges::view::transform(getName)
+		) | ::ranges::to_vector;
+		const auto i = ::ranges::find(
 			devices,
 			Global::CallOutputDeviceID(),
 			getId);
@@ -187,11 +187,11 @@ void Calls::setupContent(not_null<Window
 		st::settingsButton
 	)->addClickHandler([=] {
 		const auto &devices = VoIP::EnumerateAudioInputs();
-		const auto options = ranges::view::concat(
-			ranges::view::single(tr::lng_settings_call_device_default(tr::now)),
-			devices | ranges::view::transform(getName)
-		) | ranges::to_vector;
-		const auto i = ranges::find(
+		const auto options = ::ranges::view::concat(
+			::ranges::view::single(tr::lng_settings_call_device_default(tr::now)),
+			devices | ::ranges::view::transform(getName)
+		) | ::ranges::to_vector;
+		const auto i = ::ranges::find(
 			devices,
 			Global::CallInputDeviceID(),
 			getId);
