diff -up tdesktop-3.1.5-full/cmake/external/CMakeLists.txt.2~ tdesktop-3.1.5-full/cmake/external/CMakeLists.txt
--- tdesktop-3.1.5-full/cmake/external/CMakeLists.txt.2~	2021-10-04 14:50:25.000000000 +0200
+++ tdesktop-3.1.5-full/cmake/external/CMakeLists.txt	2021-10-04 14:53:57.278436800 +0200
@@ -37,9 +37,6 @@ if (add_hunspell_library)
 	add_checked_subdirectory(hunspell)
 endif()
 add_checked_subdirectory(iconv)
-if (LINUX)
-    add_checked_subdirectory(jemalloc)
-endif()
 add_checked_subdirectory(jpeg)
 if (LINUX AND NOT DESKTOP_APP_DISABLE_WAYLAND_INTEGRATION)
     add_checked_subdirectory(kwayland)
diff -up tdesktop-3.1.5-full/cmake/options_linux.cmake.2~ tdesktop-3.1.5-full/cmake/options_linux.cmake
--- tdesktop-3.1.5-full/cmake/options_linux.cmake.2~	2021-10-04 14:50:25.000000000 +0200
+++ tdesktop-3.1.5-full/cmake/options_linux.cmake	2021-10-04 14:53:57.279436821 +0200
@@ -52,10 +52,10 @@ if (DESKTOP_APP_SPECIAL_TARGET)
     target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
 endif()
 
-target_link_libraries(common_options
-INTERFACE
-    desktop-app::external_jemalloc
-)
+#target_link_libraries(common_options
+#INTERFACE
+#    desktop-app::external_jemalloc
+#)
 
 if (DESKTOP_APP_USE_ALLOCATION_TRACER)
     target_link_options(common_options
diff -up tdesktop-3.1.5-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp.2~ tdesktop-3.1.5-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp
--- tdesktop-3.1.5-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp.2~	2021-10-04 14:47:25.000000000 +0200
+++ tdesktop-3.1.5-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp	2021-10-04 14:53:57.279436821 +0200
@@ -47,12 +47,6 @@ https://github.com/telegramdesktop/tdesk
 
 #include <private/qguiapplication_p.h>
 
-#ifdef Q_OS_FREEBSD
-#include <malloc_np.h>
-#else // Q_OS_FREEBSD
-#include <jemalloc/jemalloc.h>
-#endif // Q_OS_FREEBSD
-
 #ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
 #include <glibmm.h>
 #include <giomm.h>
@@ -662,7 +656,6 @@ namespace Platform {
 
 void start() {
 	auto backgroundThread = true;
-	mallctl("background_thread", nullptr, nullptr, &backgroundThread, sizeof(bool));
 
 	// Prevent any later calls into setlocale() by Qt
 	QCoreApplicationPrivate::initLocale();
