diff -up tdesktop-3.4.4-full/cmake/external/CMakeLists.txt.2~ tdesktop-3.4.4-full/cmake/external/CMakeLists.txt
--- tdesktop-3.4.4-full/cmake/external/CMakeLists.txt.2~	2022-01-15 04:01:30.000000000 +0100
+++ tdesktop-3.4.4-full/cmake/external/CMakeLists.txt	2022-01-15 04:01:47.276217393 +0100
@@ -40,9 +40,6 @@ if (add_hunspell_library)
 	add_checked_subdirectory(hunspell)
 endif()
 add_checked_subdirectory(iconv)
-if (LINUX)
-    add_checked_subdirectory(jemalloc)
-endif()
 add_checked_subdirectory(jpeg)
 if (LINUX AND NOT DESKTOP_APP_DISABLE_WAYLAND_INTEGRATION)
     add_checked_subdirectory(kwayland)
diff -up tdesktop-3.4.4-full/cmake/options_linux.cmake.2~ tdesktop-3.4.4-full/cmake/options_linux.cmake
--- tdesktop-3.4.4-full/cmake/options_linux.cmake.2~	2022-01-15 04:01:47.276217393 +0100
+++ tdesktop-3.4.4-full/cmake/options_linux.cmake	2022-01-15 04:49:18.016145319 +0100
@@ -65,7 +65,7 @@ endif()
 
 target_link_libraries(common_options
 INTERFACE
-    desktop-app::external_jemalloc
+#    desktop-app::external_jemalloc
     ${CMAKE_DL_LIBS}
 )
 
diff -up tdesktop-3.4.4-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp.2~ tdesktop-3.4.4-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp
--- tdesktop-3.4.4-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp.2~	2022-01-15 04:00:49.000000000 +0100
+++ tdesktop-3.4.4-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp	2022-01-15 04:01:47.277217398 +0100
@@ -42,12 +42,6 @@ https://github.com/telegramdesktop/tdesk
 
 #include <private/qguiapplication_p.h>
 
-#ifdef Q_OS_FREEBSD
-#include <malloc_np.h>
-#else // Q_OS_FREEBSD
-#include <jemalloc/jemalloc.h>
-#endif // Q_OS_FREEBSD
-
 #ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
 #include <glibmm.h>
 #include <giomm.h>
@@ -691,7 +685,6 @@ namespace Platform {
 
 void start() {
 	auto backgroundThread = true;
-	mallctl("background_thread", nullptr, nullptr, &backgroundThread, sizeof(bool));
 
 	// Prevent any later calls into setlocale() by Qt
 	QCoreApplicationPrivate::initLocale();
