diff -up Libraries/tg_owt-master/cmake/init_target.cmake.omv~ Libraries/tg_owt-master/cmake/init_target.cmake
--- Libraries/tg_owt-master/cmake/init_target.cmake.omv~	2020-08-28 05:02:05.119965234 +0200
+++ Libraries/tg_owt-master/cmake/init_target.cmake	2020-08-28 05:05:13.452822383 +0200
@@ -76,10 +76,12 @@ function(init_target target_name) # init
             endif()
         endif()
 
-        target_compile_options(${target_name}
-        PRIVATE
-            -msse2
-        )
+	if (CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)|(amd64)|(AMD64)")
+            target_compile_options(${target_name}
+            PRIVATE
+                -msse2
+            )
+        endif()
         target_compile_definitions(${target_name}
         PRIVATE
             HAVE_NETINET_IN_H
