diff --git a/tg_owt-master/CMakeLists.txt b/tg_owt-master/CMakeLists.txt
index 58e65a06..5cedfd8f 100644
--- a/tg_owt-master/CMakeLists.txt
+++ b/tg_owt-master/CMakeLists.txt
@@ -40,12 +40,10 @@ include(cmake/init_target.cmake)
 include(cmake/target_yasm_sources.cmake)
 include(cmake/external.cmake)
 include(cmake/libabsl.cmake)
-include(cmake/libopenh264.cmake)
 include(cmake/libpffft.cmake)
 include(cmake/librnnoise.cmake)
 include(cmake/libsrtp.cmake)
 include(cmake/libusrsctp.cmake)
-include(cmake/libvpx.cmake)
 include(cmake/libwebrtcbuild.cmake)
 include(cmake/libyuv.cmake)
 if (NOT WIN32 AND NOT APPLE)
@@ -72,18 +70,12 @@ if (is_x86 OR is_x64)
         tg_owt::libwebrtcbuild
     PRIVATE
         tg_owt::libabsl
-        tg_owt::libopenh264
+        openh264
         tg_owt::libpffft
         tg_owt::librnnoise
         tg_owt::libsrtp
         tg_owt::libusrsctp
-        tg_owt::libvpx
-        tg_owt::libvpx_mmx
-        tg_owt::libvpx_sse2
-        tg_owt::libvpx_ssse3
-        tg_owt::libvpx_sse4
-        tg_owt::libvpx_avx
-        tg_owt::libvpx_avx2
+        vpx
         tg_owt::libyuv
     )
 else()
@@ -92,12 +84,12 @@ else()
         tg_owt::libwebrtcbuild
     PRIVATE
         tg_owt::libabsl
-        tg_owt::libopenh264
+	openh264
         tg_owt::libpffft
         tg_owt::librnnoise
         tg_owt::libsrtp
         tg_owt::libusrsctp
-        tg_owt::libvpx
+	vpx
         tg_owt::libyuv
     )
 endif()
@@ -109,12 +101,6 @@ if (TG_OWT_USE_PROTOBUF)
     )
 endif()
 
-target_sources(tg_owt
-PRIVATE
-    ${libvpx_yasm_objects}
-    ${libopenh264_yasm_objects}
-)
-
 if (NOT WIN32 AND NOT APPLE)
     target_link_libraries(tg_owt PRIVATE tg_owt::libevent)
 endif()
@@ -1851,28 +1837,13 @@ elseif (APPLE)
     )
 endif()
 
-set(vpx_export)
-if (is_x86 OR is_x64)
-    set(vpx_export
-        libvpx_mmx
-        libvpx_sse2
-        libvpx_ssse3
-        libvpx_sse4
-        libvpx_avx
-        libvpx_avx2
-    )
-endif()
-
 set(export_targets
     tg_owt
     libabsl
-    libopenh264
     libpffft
     librnnoise
     libsrtp
     libusrsctp
-    libvpx
-    ${vpx_export}
     libwebrtcbuild
     libyuv
     ${platform_export}
diff --git a/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc b/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc
index ffe852b7..6a9da710 100644
--- a/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc
+++ b/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc
@@ -29,10 +29,10 @@
 #include "system_wrappers/include/metrics.h"
 #include "third_party/libyuv/include/libyuv/convert.h"
 #include "third_party/libyuv/include/libyuv/scale.h"
-#include "third_party/openh264/src/codec/api/svc/codec_api.h"
-#include "third_party/openh264/src/codec/api/svc/codec_app_def.h"
-#include "third_party/openh264/src/codec/api/svc/codec_def.h"
-#include "third_party/openh264/src/codec/api/svc/codec_ver.h"
+#include <wels/codec_api.h>
+#include <wels/codec_app_def.h>
+#include <wels/codec_def.h>
+#include <wels/codec_ver.h>
 
 namespace webrtc {
 
diff --git a/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h b/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h
index 7fb87eae..59f4ae72 100644
--- a/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h
+++ b/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h
@@ -25,7 +25,7 @@
 #include "common_video/h264/h264_bitstream_parser.h"
 #include "modules/video_coding/codecs/h264/include/h264.h"
 #include "modules/video_coding/utility/quality_scaler.h"
-#include "third_party/openh264/src/codec/api/svc/codec_app_def.h"
+#include <wels/codec_app_def.h>
 
 class ISVCEncoder;
 
