diff -up tdesktop-2.4.2-full/tg_owt-master/CMakeLists.txt.1~ tdesktop-2.4.2-full/tg_owt-master/CMakeLists.txt
--- tdesktop-2.4.2-full/tg_owt-master/CMakeLists.txt.1~	2020-09-29 17:52:14.000000000 +0200
+++ tdesktop-2.4.2-full/tg_owt-master/CMakeLists.txt	2020-10-06 23:21:13.201061104 +0200
@@ -34,12 +34,10 @@ include(cmake/init_target.cmake)
 include(cmake/target_yasm_sources.cmake)
 include(cmake/external.cmake)
 include(cmake/libabsl.cmake)
-include(cmake/libopenh264.cmake)
 include(cmake/libpffft.cmake)
 include(cmake/librnnoise.cmake)
 include(cmake/libsrtp.cmake)
 include(cmake/libusrsctp.cmake)
-include(cmake/libvpx.cmake)
 include(cmake/libwebrtcbuild.cmake)
 include(cmake/libyuv.cmake)
 if (NOT WIN32 AND NOT APPLE)
@@ -58,18 +56,12 @@ if (is_x86 OR is_x64)
         tg_owt::libwebrtcbuild
     PRIVATE
         tg_owt::libabsl
-        tg_owt::libopenh264
+        openh264
         tg_owt::libpffft
         tg_owt::librnnoise
         tg_owt::libsrtp
         tg_owt::libusrsctp
-        tg_owt::libvpx
-        tg_owt::libvpx_mmx
-        tg_owt::libvpx_sse2
-        tg_owt::libvpx_ssse3
-        tg_owt::libvpx_sse4
-        tg_owt::libvpx_avx
-        tg_owt::libvpx_avx2
+        vpx
         tg_owt::libyuv
     )
 else()
@@ -78,22 +70,16 @@ else()
         tg_owt::libwebrtcbuild
     PRIVATE
         tg_owt::libabsl
-        tg_owt::libopenh264
+        openh264
         tg_owt::libpffft
         tg_owt::librnnoise
         tg_owt::libsrtp
         tg_owt::libusrsctp
-        tg_owt::libvpx
+        vpx
         tg_owt::libyuv
     )
 endif()
 
-target_sources(tg_owt
-PRIVATE
-    ${libvpx_yasm_objects}
-    ${libopenh264_yasm_objects}
-)
-
 if (NOT WIN32 AND NOT APPLE)
     target_link_libraries(tg_owt PRIVATE tg_owt::libevent)
 endif()
@@ -1819,29 +1805,14 @@ elseif (APPLE)
     )
 endif()
 
-set(vpx_export)
-if (is_x86 OR is_x64)
-    set(vpx_export
-        libvpx_mmx
-        libvpx_sse2
-        libvpx_ssse3
-        libvpx_sse4
-        libvpx_avx
-        libvpx_avx2
-    )
-endif()
-
 export(
 TARGETS
     tg_owt
     libabsl
-    libopenh264
     libpffft
     librnnoise
     libsrtp
     libusrsctp
-    libvpx
-    ${vpx_export}
     libwebrtcbuild
     libyuv
     ${platform_export}
diff -up tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc.1~ tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc
--- tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc.1~	2020-09-29 17:52:14.000000000 +0200
+++ tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.cc	2020-10-06 23:19:40.545972851 +0200
@@ -29,10 +29,10 @@
 #include "system_wrappers/include/metrics.h"
 #include "third_party/libyuv/include/libyuv/convert.h"
 #include "third_party/libyuv/include/libyuv/scale.h"
-#include "third_party/openh264/src/codec/api/svc/codec_api.h"
-#include "third_party/openh264/src/codec/api/svc/codec_app_def.h"
-#include "third_party/openh264/src/codec/api/svc/codec_def.h"
-#include "third_party/openh264/src/codec/api/svc/codec_ver.h"
+#include <wels/codec_api.h>
+#include <wels/codec_app_def.h>
+#include <wels/codec_def.h>
+#include <wels/codec_ver.h>
 
 namespace webrtc {
 
diff -up tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h.1~ tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h
--- tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h.1~	2020-09-29 17:52:14.000000000 +0200
+++ tdesktop-2.4.2-full/tg_owt-master/src/modules/video_coding/codecs/h264/h264_encoder_impl.h	2020-10-06 23:19:40.545972851 +0200
@@ -25,7 +25,7 @@
 #include "common_video/h264/h264_bitstream_parser.h"
 #include "modules/video_coding/codecs/h264/include/h264.h"
 #include "modules/video_coding/utility/quality_scaler.h"
-#include "third_party/openh264/src/codec/api/svc/codec_app_def.h"
+#include <wels/codec_app_def.h>
 
 class ISVCEncoder;
 
